src/pages/docs/tutorials/web-midi.md

```diff
@@ -0,0 +1,349 @@
+# Web MIDI
+
+This tutorial explains the basics of controlling an Elementary Audio app with the [Web MIDI API](https://developer.mozilla.org/en-US/docs/Web/API/Web_MIDI_API). The code discussed in this tutorial is available in the MIT-licensed [elementary-webmidi-demo](https://github.com/bgins/elementary-webmidi-demo) repository.
+
+## Playing notes
+We will create a simple web app that plays notes sent by a MIDI controller. The app will:
+
+- Detect when a MIDI device is connected and disconnected
+- Provide an interface for selecting a MIDI device
+- Display the last played MIDI note and its associated frequency
+
+The app uses vanilla JavaScript and no framework to make it broadly accessible. Keep in mind this is a toy application intended to convey ideas. Your app will likely use a framework, stronger types, defensive programming, and other niceties.
+
+At a high level, the app has three parts:
+
+- A Web MIDI module for interfacing with devices
+- Audio engine and synth modules for generating audio graphs and rendering them
+- An event emitter that sends note events from Web MIDI to the audio implementation
+
+The [Web Audio API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API) ultimately renders audio because we are working in the browser.
+
+## Control
+
+Our app uses [WEBMIDI.js](https://webmidijs.org/) to access the Web MIDI API. This library abstracts away many low-level Web MIDI details and makes detecting devices and capturing note events easy.
+
+The [midi module](https://github.com/bgins/elementary-webmidi-demo/blob/main/src/midi.js) includes a `noteEmitter` to send note events and `selectedInput` to track the active MIDI controller. We set the `selectedInput` to `null` until a device is selected.
+
+```JavaScript
+import { WebMidi } from "webmidi";
+
+export class Midi {
+  noteEmitter;
+  selectedInput;
+
+  constructor(noteEmitter) {
+    this.noteEmitter = noteEmitter;
+    this.selectedInput = null;
+  }
+
+  // Class methods...
+}
+```
+
+Accessing MIDI devices requires an async call and must be made in a secure context (`localhost` or a site with properly configured `https`). In addition, some browsers will prompt users for permission when a site first requests Web MIDI access.
+
+The `Midi` class has an `initialize` method that enables Web MIDI and sets up event handlers to update available MIDI devices.
+
+```JavaScript
+async initialize(displayControllers) {
+  try {
+    await WebMidi.enable();
+    displayControllers(this.#getInputNames(), this.selectedInput?.name);
```

Comment:
TIL that `#` for private access is now formally part of the javascript language itself

Reply:
Yeah! I have qualms with the syntax the went with, but it's specified and broadly available in browsers. 😄